### PR TITLE
improve zlib for nodejs_compat

### PR DESCRIPTION
**Pull request by anonrig**
Improves nodejs compat and adds a test that fails without the changes in the pull-request.

### CODE REVIEW

**Review by jasnell — APPROVED** [review]
looks like some lint issues and I'm not sure why the scrypt test is suddenly timing out but that appears unrelated. Will be investigating that separately in a bit.

File: src/workerd/api/node/zlib-util.h
Comment (npaun) [inline src/workerd/api/node/zlib-util.h:235]: Give this a try:

JSG_METHOD_NAMED(write, template write_<true>)
JSG_METHOD_NAMED(writeSync, write_<false>)

**Review by npaun — APPROVED** [review]

File: src/workerd/api/node/zlib-util.h
Comment (anonrig) [reply src/workerd/api/node/zlib-util.h:235]: unfortunately, it also doesn't work

Comment (npaun) [reply src/workerd/api/node/zlib-util.h:243]: Ah you also need an explicit instantiation of the template, to tell C++ to actually generate the code. Looks kind of clunky but it would be something like:

```c++
template void ZlibUtil::ZlibStream::write_<true>(all your params here);
template void ZlibUtil::ZlibStream::write_<false>(all your params here);
```

Comment (anonrig) [reply src/workerd/api/node/zlib-util.h:243]: yeah, i'll leave it to a different pull-request for now.

**Comment by jasnell** [comment]
Since this touches `buffersource.h`, please run an internal CI before merging.

**Comment by anonrig** [comment]
> Since this touches `buffersource.h`, please run an internal CI before merging.

Internal PR passes.

**Pull request merged**
